src/hooks/boardwalk/useWallet.tsx

```diff
@@ -0,0 +1,172 @@
+    * @param contact Contact to lock the ecash too and send a notification to
+    * @returns If succesfful and not sending to a mintless user it returns sent data, otherwise undefined (including on error)
+    */
+   const sendEcash = async (
+      amount: number,
+      unit: Currency,
+      onSendToMintlessUser: () => void,
+      gift?: GiftAsset,
+      contact?: PublicContact,
+   ): Promise<
+      | {
+           token: string;
+           txid: string | undefined;
+           gift: GiftAsset | undefined;
+        }
+      | undefined
+   > => {
+      let token: string | undefined;
+      try {
+         if (isMintless && !contact?.mintlessReceive) {
+            /* if we are making a lightning payment to a user with a mint */
+            if (!contact) {
+               addToast('You can only send eTips and eGifts from a Lightning Wallet.', 'error');
+               throw new Error('You can only send eTips and eGifts from a Lightning Wallet.');
+            }
+            if (unit !== 'sat') {
+               throw new Error('mintless only supports sat');
+            }
+
+            token = await createMintlessToken(Math.round(amount), unit, contact);
+         } else if (contact?.mintlessReceive) {
+            /* user wants to receive to their lud16 */
+            if (!contact.lud16) {
+               addToast('Contact does not have a lightning address', 'error');
+               return;
+            }
+
+            await sendToMintlessUser(amount, activeUnit, contact);
+            onSendToMintlessUser();
+         } else {
+            /* regular send */
+            console.log('creating token for ', amount);
+            token = await createSendableToken(amount, {
+               pubkey: contact ? `02${contact.pubkey}` : undefined,
+               gift: gift?.name,
+               feeCents: gift?.fee,
+            });
+         }
+
+         if (!token) {
+            throw new Error('Failed to create ecash token');
+         }
+
+         let txid: string | undefined;
+         if (contact) {
+            await sendTokenAsNotification(token);
+            txid = await postTokenToDb(token);
+         }
+
+         return { token, txid, gift };
+      } catch (e: any) {
+         console.error(e);
+      }
+   };
+
+   /**
+    * Send a gift as ecash or to a mintless user
+    * @param gift [`GiftAsset`] to send
+    * @param onSendToMintlessUser Called when sending to a mintless user. Should reset the parent component's state
+    * @param contact Optional contact to send the gift to
+    * @returns If succesfful and not sending to a mintless user it returns sent data, otherwise undefined (including on error)
+    */
+   const sendGift = async (
+      gift: GiftAsset,
+      onSendToMintlessUser: () => void,
+      contact?: PublicContact,
+   ): Promise<{ txid?: string; token?: string } | undefined> => {
+      try {
+         if (gift?.campaingId) {
+            throw new Error('Gift campaigns not implemented');
```

Comment:
campaigns are a feature we tried out where Boardwalk could sponsor gifts by basically subsidizing the first x amount of users to send a "Campaign Gift"